### PR TITLE
Use ForEachActiveObject to iterate system pool object

### PR DESCRIPTION
**Pull request by yufengwangca**
#### Problem
What is being fixed?  Examples:
* We should stop using inclusive forloop to iterate the system pool object since we have supported CHIP_SYSTEM_CONFIG_POOL_USE_HEAP configuration now. The system pool size is no longer fixed under this configuration.

#### Change overview
Use ForEachActiveObject to iterate system pool

#### Testing
How was this tested? (at least one bullet point required)
* Confirm TestTCP is not broken with this change.

### CODE REVIEW

**Review by saurabhst — APPROVED** [review]

File: src/inet/InetLayer.cpp
Comment (tcarmelveilleux) [inline src/inet/InetLayer.cpp:984]: Why the change to `return true` from `continue` ? This changes the semantics of the loop. Not sure of the original semantics (sure could use some comments), but is it intended?

Comment (yufengwangca) [reply src/inet/InetLayer.cpp:984]: Return true if you want to skip to the next iteration in the ForEachActiveObject loop, return false if you want to jump out of ForEachActiveObject loop.

**Review by yunhanw-google — APPROVED** [review]

File: src/inet/InetLayer.cpp
Comment (tcarmelveilleux) [inline src/inet/InetLayer.cpp:978]: Please:

- Use explicit capture by reference of just `lEndpoint`
- Make explicit the return type of the lambda (` -> bool`) so that the `return true` inner is more obvious. It's not necessarily obvious to the reader who has never seen `ForEachActiveObject` API what the return value means (e.g. that `return true` means continue iterating.

**Review by tcarmelveilleux — APPROVED** [review]
Approved pending addressing comment https://github.com/project-chip/connectedhomeip/pull/9198/files#r694861426

**Review by woody-apple — APPROVED** [review]

**Comment by msandstedt** [comment]
Do we still have plans to remove the indexed getter and size APIs?  Is this blocked on test code refactors?

**Comment by yufengwangca** [comment]
> Do we still have plans to remove the indexed getter and size APIs? Is this blocked on test code refactors?

That is my plan, have the draft version ready and will post PR after this one get merged

File: src/inet/InetLayer.cpp
Comment (yufengwangca) [reply src/inet/InetLayer.cpp:978]: Acked, this change needs to apply to all other occurrence of ForEachActiveObject throughout the stack. https://github.com/project-chip/connectedhomeip/issues/9209 to follow up this refactor.

**Review by msandstedt — APPROVED** [review]

**Comment by woody-apple** [comment]
@saurabhst @Damian-Nordic ?

**Review by saurabhst — APPROVED** [review]

**Comment by woody-apple** [comment]
@mspang ?

**Review by andy31415 — APPROVED** [review]

**Pull request merged**
